node/channel.js

```diff
@@ -721,17 +720,18 @@ TChannel.prototype._request = function _request(opts) {
         req = opts.peer.request(opts);
     } else if (opts.host) {
         opts.retryCount = 0;
-        req = self.peers.add(opts.host).request(opts);
+        opts.peer = self.peers.add(opts.host);
```

Comment:
I am considering a follow-up where the channel and peer are required positional arguments for Request instead of “required options”.